### PR TITLE
Authentication with personal access token

### PR DESCRIPTION
**Pull request by palyuga**
A support of personal access token is added. Unlike password, token is persisted between sessions so used doesn't have to enter anything after IDE starts. Now there are two request factories: for basic auth and access token auth.
Readme.md is also updated to reflect the changes.

Issue #66

### CODE REVIEW

**Comment by mattdillon100** [comment]
Can we please get this merged?!

**Review by a-krasilnikov — APPROVED** [review]

**Comment by palyuga** [comment]
@wildcard27 a new version that supports access token auth is waiting for JetBrains' approval

**Pull request merged**
